lib/router.js

```diff
@@ -46,10 +46,26 @@ export default class Router {
 
     this.historyMethod = historyMethod;
     this.origChangeStateMethod = this.window.history[this.historyMethod];
-    this.window.history[this.historyMethod] = (...args) => {
-      this.origChangeStateMethod.apply(this.window.history, args);
-      this.navigateToHash();
-    };
+
+    // If this is the true *original* native method, fire pushstate/replacestate events so
+    // that other components can be notified of history changes. Subsequent panel routers
+    // will re-wrap the method but not fire duplicate pushstate/replacestate events.
+    // Note: this.origChangeStateMethod.name === this.historyMethod is an effective means
+    // of verifying whether this.window.history[this.historyMethod] has been wrapped yet or
+    // not because the original native function will have name === "pushState" / "replaceState",
+    // but the wrapped version will have name === undefined since it is anonymous.
+    if (this.origChangeStateMethod.name === this.historyMethod) {
+      this.window.history[this.historyMethod] = (...args) => {
+        this.origChangeStateMethod.apply(this.window.history, args);
+        this.navigateToHash();
+        window.dispatchEvent(new CustomEvent(this.historyMethod.toLowerCase()));
```

Comment:
in other words:
```suggestion
        this.app.dispatchEvent(new CustomEvent(this.historyMethod.toLowerCase()));
```

Reply:
I'm definitely on board with this, but how do you feel about adding `{bubbles: true}` to the event? Then (in the concrete example) mp-chrome can add a listener to document instead of getting a reference to the Lexicon element to add the listener. If mp-chrome relies on access to lexicon the order they're initialized becomes important which doesn't seem great - I believe lexicon is initialized 2nd and it seems like we don't want order to matter.

I think the greater issue here is communication between components that don't have an ancestor->descendant relationship in the dom. In this case they both want to do something when url changes - lexicon navigates to a new route, mp-chrome highlights a new report tab - so this event dispatching seems better than having an ancestor component of both receive an explicit event from lexicon and update mp-chrome accordingly. That does seem like the "right" way, but would be so much work when we want such a simple end result here (update component on url change)

Reply:
would this not work for your case?
```js
const app = document.createElement(`lexicon-app`);
app.addEventListener(`pushstate`, () => window.dispatchEvent(new CustomEvent(`pushstate`)));
```

Reply:
+1 on bubbles. Makes it easier to track all navigation changes in one place.

Reply:
that wfm yeah. bubbling would be convenient but I can see why that's cleaner as the lib default.